### PR TITLE
Fix #1464 - alt display for SV, STR GQ. Parse STR depth.

### LINKED ISSUE #1464: Quality scores for SVs
As a user of Scout, I would like SV quality scores to be clearly represented and the scale documented. -1 is not a great way to report SV GQ. Likely present on the VCF but may require specialized parsing compared to the SNV GQs.

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes an issue.

SV and STR GQ were generally left on "-1" as several SV callers and the current STR caller did not produce a gq variant format tag. Replacing display with other relevant info. Noticed in the same display card that allele depth was also not well parsed for STRs, so added in that as well.
 
**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by dnil** [comment]
So far, so good:
Screenshot 2021-02-09 at 11 55 44

**Comment by dnil** [comment]
Ok, almost there for the first pass. If the automatic tests pan out I think I will go test this on stage to have it parse a bit wilder data than the current demo. And write a couple of more tests; the coverage may have dropped slightly. 😊

**Comment by dnil** [comment]
The power of better example files I guess. Ready for review then.. 😸

**Comment by northwestwitch** [comment]
Nice @dnil ! I can review tomorrow if someone else isn't faster

File: scout/parse/variant/genotype.py [hunk 1]
```diff
@@ -189,6 +227,15 @@ def parse_genotype(variant, ind, pos):
             if split_read_ref:
                 ref_depth += split_read_ref
 
+        if spanning_ref is not None or flanking_ref is not None or inrepeat_ref is not None:
```
Comment (northwestwitch) [inline scout/parse/variant/genotype.py:230]: if any( [spanning_ref, flanking_ref, inrepeat_ref] ):

File: scout/parse/variant/genotype.py [hunk 2]
```diff
@@ -215,3 +268,39 @@ def parse_genotype(variant, ind, pos):
...
+            if len(values) > 1:
+                ref_value = int(values[0])
+                alt_value = int(values[1])
+            if len(values) == 1:
+                alt_value = int(values[0])
+            if ref_value >= 0:
+                ref = ref_value
+            if alt_value >= 0:
+                alt = alt_value
+        except (ValueError, TypeError) as e:
```
Comment (northwestwitch) [inline scout/parse/variant/genotype.py:304]: This wouldn't cause any trouble I guess :)

File: scout/server/blueprints/variant/templates/variant/variant_details.html [hunk 3]
```diff
@@ -49,7 +59,21 @@
...
+                <td class="text-right">
+                {% if variant.category == "sv" %}
+                  {% if sample.genotype_quality != "-1" %}
+                    {{ sample.genotype_quality }} (VQ {{variant.quality}})
+                  {% elif variant.quality != "-1" %}
+                    <small>VQ</small> {{ variant.quality }}
+                  {% else %}
+                    <small>N/A</small>
+                  {% endif%}
+                {% elif variant.category == "str" %}
```
Comment (northwestwitch) [inline scout/server/blueprints/variant/templates/variant/variant_details.html:71]: 👍

File: scout/server/config.py [hunk 4]
```diff
@@ -35,6 +35,22 @@
 # connection details for LoqusDB MongoDB database in the loqusdb config file
 #    uri=("mongodb://{}:{}@localhost:{}/loqusdb".format(MONGO_USERNAME, MONGO_PASSWORD, MONGO_PORT))
 
+CLOUD_IGV_TRACKS = [
```
Comment (northwestwitch) [inline scout/server/config.py:38]: Left it there while testing some other issue?

**Review by northwestwitch — APPROVED** [review]
Works perfectly! I have just a few suggestions.

SVs quality 👍  

image


STRs on master:

image


STRs on this branch:

image

File: scout/parse/variant/genotype.py [hunk 5]
```diff
@@ -17,6 +17,9 @@
 'RV' that are number of split reads that supports the event.
 
 """
+import logging
+
+LOG = logging.getLogger(__name__)
```
[the remark this answers is not in the document]
Comment (dnil) [reply scout/parse/variant/genotype.py:22]: Indeed! 😄

File: scout/parse/variant/genotype.py (hunk 2 above)
Comment (dnil) [reply scout/parse/variant/genotype.py:304]: 😆 Nooo! How could it ever? NoneType is a mixed blessing..

File: scout/server/config.py (hunk 4 above)
Comment (dnil) [reply scout/server/config.py:38]: Indeed! But actually, let me just comment it out like the other in-config examples perhaps?

**Comment by dnil** [comment]
Adjusted slightly based on stage test with a couple of more variant callers; in particular TIDDIT and CNVnator.
Previous version of this branch:
Screenshot 2021-02-10 at 11 07 18
Now:
Screenshot 2021-02-10 at 11 16 41

**Pull request merged**